sdk/core/azure-core-tracing-opentelemetry/azure/core/tracing/ai/inference/_ai_inference_api_instrumentor_impl.py

```diff
@@ -0,0 +1,473 @@
+    if item.delta.content:
+        accumulate.setdefault("message", {})
+        accumulate["message"].setdefault("content", "")
+        accumulate["message"]["content"] += item.delta.content
+    if item.delta.tool_calls:
+        accumulate.setdefault("message", {})
+        accumulate["message"].setdefault("tool_calls", [])
+        if item.delta.tool_calls is not None:
+            for tool_call in item.delta.tool_calls:
+                if tool_call.id:
+                    accumulate["message"]["tool_calls"].append({"id": tool_call.id, "type": "", "function": {"name": "", "arguments": ""}})
+                if tool_call.function:
+                    accumulate["message"]["tool_calls"][-1]["type"] = "function"
+                if tool_call.function and tool_call.function.name:
+                    accumulate["message"]["tool_calls"][-1]["function"]["name"] = tool_call.function.name
+                if tool_call.function and tool_call.function.arguments:
+                    accumulate["message"]["tool_calls"][-1]["function"]["arguments"] += tool_call.function.arguments
+
+
+def _wrapped_stream(stream_obj: _models.StreamingChatCompletions, span: AbstractSpan) ->  _models.StreamingChatCompletions:
+    class StreamWrapper(_models.StreamingChatCompletions):
+        def __init__(self, stream_obj):
+            super().__init__(stream_obj._response)
+
+        def __iter__(self) -> Iterator[_models.StreamingChatCompletionsUpdate]:
+            global _trace_inference_content
+            try:
+                accumulate: Dict[str, Any] = {}
+                for chunk in stream_obj:
+                    for item in chunk.choices:
+                        _accumulate_response(item, accumulate)
+                    yield chunk
+
+                _add_response_chat_attributes(span, chunk)
+
+            except Exception as exc:
+                # Set the span status to error
+                if isinstance(span.span_instance, Span):
+                    span.span_instance.set_status(StatusCode.ERROR, description=str(exc))
+                module = exc.__module__ if exc.__module__ != "builtins" else ""
+                error_type = f"{module}.{type(exc).__name__}" if module else type(exc).__name__
+                _set_attributes(span, ("error.type", error_type))
+                raise
+
+            finally:
+                if stream_obj._done is False:
+                    if accumulate.get("finish_reason") is None:
+                        accumulate["finish_reason"] = "error"
+                else:
+                    # Only one choice expected with streaming
+                    accumulate["index"] = 0
+                    # Delete message if content tracing is not enabled
+                    if not _trace_inference_content:
+                        if 'message' in accumulate:
+                            if 'content' in accumulate['message']:
+                                del accumulate['message']['content']
```

Comment:
could we avoid collecting information if content is not enabled to save some perf?

Reply:
I created a work item for investigating this.